### ISSUE TITLE
How to save model to best iteration if early stopping does not happen?

### ISSUE THREAD

**Issue by kn45** [Microsoft/LightGBM#1683]
If early stopping not happens, save_model() save all the trees and model.best_iteration is the last iter. However, there is a best iteration so far (not the last one). How to save model up to this best iteration?

**Comment by guolinke** [comment]
I think it will auto save the best iteration: https://github.com/Microsoft/LightGBM/blob/master/python-package/lightgbm/basic.py#L1919-L1946
ping @StrikerRUS to confirm.

**Comment by StrikerRUS** [comment]
@guolinke Yeah, it saves `best_iteration` by default.

@kn45 Do you set `best_iteration` manually? Or why do you have best iteration without early stopping occurrence? Please describe your use case in more details.

**Comment by kn45** [comment]
Supposing I set early_stopping_rounds=20 num_tree=100, on the validation set No.90 iteration got the best AUC so I want to save the model with 90 trees.
However, `best_iteration` in save_model() is 100 because early_stopping not happens and save_model() would dump all the 100 trees.

**Comment by guolinke** [comment]
@kn45 You can try the latest version, even if the early_stopping not happens, the best_iteration is 90 in your case.

**Comment by StrikerRUS** [comment]
@kn45 If there was no early stopping, then you can directly specify the needed iteration, like:
```
booster.save_model(filename='path_to_the_model', num_iteration=90)
```

**Comment by kn45** [comment]
ok, i'ill try the latest version.
